lighthouse-core/test/fraggle-rock/api-test-pptr.js

```diff
@@ -66,19 +87,49 @@ describe('Fraggle Rock API', () => {
       const accessibility = lhr.categories.accessibility;
       expect(accessibility.score).toBeLessThan(1);
 
-      const auditResults = accessibility.auditRefs.map(ref => lhr.audits[ref.id]);
-      const irrelevantDisplayModes = new Set(['notApplicable', 'manual']);
-      const applicableAudits = auditResults
-        .filter(audit => !irrelevantDisplayModes.has(audit.scoreDisplayMode));
+      const {auditResults, erroredAudits, failedAudits} = getAuditsBreakdown(lhr);
+      // TODO(FR-COMPAT): This assertion can be removed when full compatibility is reached.
+      expect(auditResults.length).toMatchInlineSnapshot(`58`);
+
+      expect(erroredAudits).toHaveLength(0);
+      expect(failedAudits.map(audit => audit.id)).toContain('label');
+    });
+  });
+
+  describe('startTimespan', () => {
+    beforeEach(() => {
+      server.baseDir = path.join(__dirname, '../fixtures/fraggle-rock/snapshot-basic');
+    });
+
+    it('should compute ConsoleMessage results across a span of time', async () => {
+      const run = await lighthouse.startTimespan({page});
+
+      await setupTestPage();
+
+      const result = await run.endTimespan();
+      if (!result) throw new Error('Lighthouse failed to produce a result');
+
+      const {lhr} = result;
+      const bestPractices = lhr.categories['best-practices'];
+      expect(bestPractices.score).toBeLessThan(1);
+
+      const {erroredAudits, failedAudits} = getAuditsBreakdown(lhr);
 
-      const erroredAudits = applicableAudits
-        .filter(audit => audit.score === null);
       expect(erroredAudits).toHaveLength(0);
+      expect(failedAudits.map(audit => audit.id)).toContain('errors-in-console');
 
-      const failedAuditIds = applicableAudits
-        .filter(audit => audit.score !== null && audit.score < 1)
-        .map(audit => audit.id);
-      expect(failedAuditIds).toContain('label');
+      const errorsInConsole = lhr.audits['errors-in-console'];
+      if (!errorsInConsole.details) throw new Error('Error in consoles audit missing details');
+      if (errorsInConsole.details.type !== 'table') throw new Error('Unexpected details');
+      const errorLogs = errorsInConsole.details.items;
+      const matchingLog = errorLogs.find(
+        log =>
+          log.source === 'console.error' &&
+          String(log.description || '').includes('violations added')
```

Comment:
```suggestion
         (log.description || '').includes('violations added')
```

same?

Reply:
or

```suggestion
          log.description && log.description.includes('violations added')
```

Reply:
unfortunately doesn't work. `description` is an item details which is typed `Record<string, a|bunch|of|junk>` . would you prefer a type assertion and new lines to the magic I have here? :)